### ISSUE TITLE
Tax-Calculator usage example: making taxcalc packages

### ISSUE THREAD

**Issue by martinholmer** [open-source-economics/policybrain-builder#74]
Here is an example of a bash shell script that uses the policybrain-builder CLI, `pb`, **release 0.1.0** to build and upload conda taxcalc packages for Python 2.7 and Python 3.6 for a specified Tax-Calculator release.

Before executing this bash script **be sure that the `conda-build` package is installed**.
If it is not installed, then execute this command: `conda install conda-build`

Also, note that in the `find` statement in the bash script you need to change `anaconda2` to `anaconda3` if you are running Python 3.6+ instead of Python 2.7.

Here is content of the bash shell script, which is called `taxcalc-packages.sh`:
```
$ pb --version
pb, version 0.1.0

$ cat taxcalc-packages.sh 
#!/bin/bash
# Script to build and upload taxcalc packages for specified release using
# policybrain-builder CLI, pb, with a single parameter, TAG.
# USAGE: pb TAG

echo "STARTING : `date`"

TOKEN=<secret-token>
OPTIONS="--token $TOKEN --clean --label main"
pb release $OPTIONS taxcalc=$1 2>&1 | awk '$1~/\[taxcalc\]/'

conda build purge
find ~/anaconda2/conda-bld -name "*tar.bz2" -exec rm -f {} \;

echo "FINISHED : `date`"
exit 0
```
Here's a recent use example:
```
$ ./taxcalc-packages.sh 0.16.0
STARTING : Tue Feb 13 18:13:33 EST 2018
[taxcalc] removing
[taxcalc] cloning
[taxcalc] fetching
[taxcalc] checking out '0.16.0'
[taxcalc] archiving
[taxcalc] building 3.6
[taxcalc] converting to linux-64
[taxcalc] converting to win-32
[taxcalc] converting to win-64
[taxcalc] building 2.7
[taxcalc] converting to linux-64
[taxcalc] converting to win-32
[taxcalc] converting to win-64
[taxcalc] uploading osx-64 packages
[taxcalc] uploading linux-64 packages
[taxcalc] uploading win-32 packages
[taxcalc] uploading win-64 packages
FINISHED : Tue Feb 13 18:31:13 EST 2018
```
Note that this script builds eight packages: four platforms (osx-64, linux-64, win-32, win-64) for each Python version (2.7, 3.6).

And finally, here is what things look like on `anaconda.org` after the bash script finished execution:

![screen shot 2018-02-14 at 2 42 38 pm](https://user-images.githubusercontent.com/12170745/36224435-972fb48a-1195-11e8-92a9-6cdf2a5855a5.png)

**Comment by MattHJensen** [comment]
@martinholmer @jbcrail thanks for the CLI and usage example. I have copied this bash shell script into ./policybrain-builder and am attempting to use it to upload packages, but am hitting an error. I am currently investigating, but if either of you have suggestions I'd appreciate them. 

```
(taxcalc-dev) bn0kg72-l:policybrain-builder Matt.Jensen$ ./taxcalc-packages.sh
./taxcalc-packages.sh: line 1: $: command not found
STARTING : Tue Nov  7 14:32:34 EST 2017
[taxcalc] removing
[taxcalc] cloning
[taxcalc] fetching
[taxcalc] checking out '0.9.2'
[taxcalc] archiving
[taxcalc] building 3.6
Error: no such directory: /Users/Matt.Jensen/proj/policybrain-builder/purge
FINISHED : Tue Nov  7 14:34:29 EST 2017
```

**Comment by martinholmer** [comment]
@MattHJensen said:
> @martinholmer @jbcrail thanks for the CLI and usage example.  I have copied this bash shell script into ./policybrain-builder and am attempting to use it to upload packages, but am hitting an error. I am currently investigating, but if either of you have suggestions I'd appreciate them.
>```
> (taxcalc-dev) bn0kg72-l:policybrain-builder Matt.Jensen$ ./taxcalc-packages.sh
>STARTING : Tue Nov  7 14:32:34 EST 2017
> [taxcalc] removing
> [taxcalc] cloning
> [taxcalc] fetching
> [taxcalc] checking out '0.9.2' . <--------------- VERY BIG PROBLEM -----------------
> [taxcalc] archiving
> [taxcalc] building 3.6
> Error: no such directory: /Users/Matt.Jensen/proj/policybrain-builder/purge
> FINISHED : Tue Nov  7 14:34:29 EST 2017

Proper usage now is: `./tax-packages.sh 0.13.0`
Isn't that the release you want to build and upload the conda packages for?

**Comment by MattHJensen** [comment]
Thanks @martinholmer for the feedback. I am getting the same error, though:

```
bn0kg72-l:policybrain-builder Matt.Jensen$ ./taxcalc-packages.sh 0.13.0
./taxcalc-packages.sh: line 1: $: command not found
STARTING : Tue Nov  7 18:03:11 EST 2017
[taxcalc] resetting
[taxcalc] pulling
[taxcalc] fetching
[taxcalc] checking out '0.13.0'
[taxcalc] archiving
[taxcalc] building 3.6
Error: no such directory: /Users/Matt.Jensen/proj/policybrain-builder/purge
FINISHED : Tue Nov  7 18:03:15 EST 2017
bn0kg72-l:policybrain-builder Matt.Jensen$ 

```

**Comment by martinholmer** [comment]
@MattHJensen reported this error:
```
[taxcalc] building 3.6
Error: no such directory: /Users/Matt.Jensen/proj/policybrain-builder/purge
```
The CLI is trying to do something with that directory but it doesn't exist.
- Have you done `python setup.py install`?
- Do you have the `conda-build` package installed?
- Have you tried `mkdir purge` so that there is an empty directory with that name?

**Comment by martinholmer** [comment]
@MattHJensen reported:
```
thanks for helping me track this down.

Have you done python setup.py install?
- Yes. `pb --version` returns pb, version 0.0.1

Do you have the conda-build package installed?
- Yes. `conda-build --version` returns conda-build 1.8.2

Have you tried mkdir purge so that there is an empty directory with that name?
- Yes. The output is then:

bn0kg72-l:policybrain-builder Matt.Jensen$ ./taxcalc-packages.sh 0.13.0
STARTING : Tue Nov  7 18:17:31 EST 2017
[taxcalc] removing
[taxcalc] cloning
[taxcalc] fetching
[taxcalc] checking out '0.13.0'
[taxcalc] archiving
[taxcalc] building 3.6
Error: meta.yaml or conda.yaml not found in /Users/Matt.Jensen/proj/policybrain-builder/purge
FINISHED : Tue Nov  7 18:19:07 EST 2017
```
This is a new error that mentions "meta.yaml or conda.yaml"

My only remaining guess is that for some reason `pb` doesn't like running in a conda environment.
I have no reason to believe that is true other than the fact that that is the only remaining difference I con remember between what you're doing and what I done many times with any of these problems.  In other words, I run `pb` without being in a conda environment.  While it seems as if you are in an environment when you run `pb`.

One other straw to grasp at: the last bugs @jbcrail fixed in `pb` had to do with the `--clean` option.
Maybe there is still a bug.  Perhaps try to remove the `--clean` option in the bash script to see if that helps.

**Comment by MattHJensen** [comment]
> My only remaining guess is that for some reason pb doesn't like running in a conda environment.
> I have no reason to believe that is true other than the fact that that is the only remaining difference I con remember between what you're doing and what I done many times with any of these problems. In other words, I run pb without being in a conda environment. While it seems as if you are in an environment when you run pb.
> 
> One other straw to grasp at: the last bugs @jbcrail fixed in pb had to do with the --clean option.
> Maybe there is still a bug. Perhaps try to remove the --clean option in the bash script to see if that helps.

Neither of these changes are working, even starting over from scratch entirely outside of a conda env. 

@martinholmer, would you be willing to do the 0.13.0 conda release so I can figure this out without blocking others?

**Comment by martinholmer** [comment]
@MattHJensen,  I just now got my Toronto iMac set up, so I can comply with your request:
> would you be willing to do the 0.13.0 conda release so I can figure this out without blocking others?

Seems to be working fine here in Canada.  I've update the initial comment in the USAGE EXAMPLE issue to reflect the latest situation.

**Comment by MattHJensen** [comment]
Thanks very much @martinholmer! Was that with a brand new install? I'll report back when I figure this out. My current hunch is that it might have something to do with bash permissions.

**Comment by martinholmer** [comment]
@MattHJensen asked:
> Was that [successful execution of `taxcalc-packages.sh` script]with a brand new install?

Yes.

**Comment by martinholmer** [comment]
Closed in favor of the newer issue #86.
